### PR TITLE
Update UG with lesson find

### PR DESCRIPTION
**Pull request by ethantyh692**

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -263,5 +285,6 @@ the data of your previous AddressBook home folder.
  **List Contacts**  | `contact list`                                                                                                                                                                
  **Add Lesson**     | `lesson add  sid/STUDENTID d/STARTDATE li/LOCATIONINDEX h/DURATION` <br> e.g., `lesson add sid/1 d/15-10-2024 23:47 li/1 h/1`                                         
  **List Lessons**   | `lesson list`                                                                                                                                                         
- **Delete Lessons** | `lesson delete INDEX` <br> e.g., `lesson delete 1`                                                                                                                    
+ **Delete Lessons** | `lesson delete INDEX` <br> e.g., `lesson delete 1`
+ **Find Lessons** | `lesson find KEYWORD [MORE_KEYWORDS]` <br> e.g., `lesson find John Alice`
```
Comment (NorbertLoh) [inline docs/UserGuide.md:289]: Perhaps you can align the | with the other rows to keep it neat,

**Review by NorbertLoh — COMMENTED** [review]
Nits to fix. Other than that LGTM!

File: docs/UserGuide.md [hunk 2]
```diff
@@ -195,6 +195,28 @@ Examples:
 
 * `lesson delete 2` deletes the 2nd lesson in the lessons schedule.
 
+### Finding lessons : `lesson find`
+
+Finds lessons whose student's name contain any of the given keywords.
```
Comment (Chronoxy) [inline docs/UserGuide.md:200]: I'm not sure if I like the phrasing of this sentence.
Maybe you can try, "
Find lessons with students whose names contain any of the specified keywords."

File: docs/UserGuide.md [hunk 3]
```diff
@@ -195,6 +195,28 @@ Examples:
...
+
+Format: `lesson find KEYWORD [MORE_KEYWORDS]`
+
+* The search is case-insensitive. e.g `hans` will match `Hans`
+* The order of the keywords does not matter. e.g. `Hans Bo` will match `Bo Hans`
+* Only the name is searched.
+* Only full words will be matched e.g. `Han` will not match `Hans`
+* Lessons whose persons matching at least one keyword will be returned (i.e. `OR` search).
```
Comment (Chronoxy) [inline docs/UserGuide.md:208]: Similar to previous comment, maybe you can try
"Lessons with students whose names match at least one of the specified keywords will be returned (i.e., an OR search)."

**Review by Chronoxy — CHANGES_REQUESTED** [review]
Some nits to check on, but LGTM overall 👍

File: docs/UserGuide.md (hunk 3 above)
Comment (ethantyh692) [reply docs/UserGuide.md]: Changed

**Review by NorbertLoh — DISMISSED** [review]
LGTM!

File: docs/UserGuide.md [hunk 4]
```diff
@@ -195,6 +195,29 @@ Examples:
...
+
+Format: `lesson find KEYWORD [MORE_KEYWORDS]`
+
+* The search is case-insensitive. e.g `hans` will match `Hans`
+* The order of the keywords does not matter. e.g. `Hans Bo` will match `Bo Hans`
+* Only the name is searched.
+* Only full words will be matched e.g. `Han` will not match `Hans`
+* Lessons with students whose names match at least one of the specified keywords will be returned (i.e., 
+  an OR search)
+  e.g. `Hans Bo` will return lessons with `Hans Gruber` if any and `Bo Yang` if any
```
Comment (NorbertLoh) [inline docs/UserGuide.md:210]: Perhaps you can end every bullet point with . to match the others.

**Review by NorbertLoh — COMMENTED** [review]
Nits to fix. Other than that LGTM!

**Review by NorbertLoh — APPROVED** [review]
LGTM!

**Pull request merged**